### ISSUE TITLE
Origin: null on iOS with cordova-plugin-ionic-webview

### ISSUE THREAD

**Issue by brassier** [Microsoft/cordova-plugin-code-push#489]
When adding the most recent code-push cordova plugin, iOS API requests are sending `Origin: null` headers when the cordova-plugin-ionic-webview plugin is used.  This used to send localhost:8080 as the origin.  A null origin is not expected, and it is also not easy to deal with for CORS setups since it's not a valid URI.

### Reproduction

 - Create a blank Ionic project
 - Have the Ionic project make an API call
 - Add the latest cordova-plugin-code-push plugin (version 1.11.16)
 - Add the cordova-plugin-ionic-webview plugin (version 1.x or 2.x or 3.x)
 - Add the cordova ios platform (4.5.0)
 - Run the project and make an API call.  Notice the *unexpected* `Origin: null` header
 - Remove the cordova-plugin-code-push plugin
 - Remove the platform & re-add it & rebuild it
 - Run the project and make an API call.  Notice the *expected* `Origin: http://localhost:8080` header

### Additional Information

* cordova-plugin-code-push version: 
  * 1.11.16
* List of installed plugins:
  * cordova-plugin-ionic-webview (1.x or 2.x or 3.x)
  * others which don't seem to be related
* Cordova version:
  * CLI 6.5.0
* iOS/Android/Windows version:
  * ios@4.5.0
* Does this reproduce on a debug build or release build?
  * debug
* Does this reproduce on a simulator, or only on a physical device?
  * only tried physical device

**Comment by AMerkuri** [comment]
I have same problem. I've updated XCode to 10.1 because apparently 9.4.1 is no longer going to be supported and work when publishing apps to app store.

Cordova 8.0.0, cordova-ios 4.5.5, cordova-plugin-ionic-webview 3.0.0, cordova-plugin-code-push 1.11.16

**Comment by hciyang** [comment]
same issue here with cordova-plugin-code-push 1.11.16
Any updates or workaround? 

XCode 10.1, Cordva 8, cordova-ios 4.5.3, cordova-plugin-ionic-webview 1.1.21.

**Comment by alexandergoncharov** [comment]
Hi @brassier ,
Thanks for reporting and sorry for delay.

Could you please clarify this more detailed: `Run the project and make an API call.`? 
What kind of call do you mean?

**Comment by brassier** [comment]
@alexandergoncharov - sorry, bad word choice.  What I meant was make an HTTP-based web API call (to any API).  The point is seeing that the outgoing HTTP request now has an `Origin: null` header which really messes with CORS setups on server-side web APIs.  In iOS this used to come through as `Origin: http://localhost:8080` (but this can be overriden with the ionic-webview settings).  If you remove the code-push plugin it goes back to `http://localhost:8080`, so code-push is stepping on that Origin header, making it `null` for some reason.

**Comment by alexandergoncharov** [comment]
@brassier, oh, I see. Thanks.
Could you please test `1.11.14` for workaround? I think it should fix your issue.
Also, could you please provide some dummy project with reproducing issue and reprosteps?

**Comment by hciyang** [comment]
I still see the same issue on 1.11.17, "Origin null is not allowed". My project was working fine when I rolled back to 1.11.15.

**Comment by alexandergoncharov** [comment]
@brassier Yeah, `1.11.16` version had issues with Ionic 4. This was fixed in `1.11.17` version.

@hciyang, Thanks for reporting! Could you please provide dummy project with reproducing issue and reprosteps for `1.11.17` version?

**Comment by brassier** [comment]
@hciyang - I personally had some challenges getting cordova to get the proper version.  For example, I explicitly installed `@1.11.17` when I already had `@1.11.16` installed, but my project stayed at `@1.11.16` (as shown with `cordova plugin list`).  Not sure if you are seeing the same.

It was most accurate for me when I actually removed the plugin, then removed the platform, then removed the `plugins` and `platforms` directory, then re-added things as needed.  Probably overkill, but it seemed to be more reliable.

**Comment by alexandergoncharov** [comment]
@brassier, yeah, the most reliable approach is removing platform with plugin. Don't forget to add platform back :)

**Comment by AMerkuri** [comment]
> Hi @alexandergoncharov - I do have a simple app that reproduces this, but you have to open safari dev tools and watch the network traffic to see the `Origin` header. A little clunky, but I can post if you want it.
> 
> However, when I was confirming this project still reproduced the issue, I couldn't get it to reproduce. I noticed the app picked up `1.11.17`. It seems like that version fixed the issue in question. If I force the project back to `1.11.16` I can recreate it, but not on `1.11.17`.
> 
> @hciyang and @AMerkuri - can you confirm the same with your apps? If so, we can likely close this as an issue specific to `1.11.16` only.

Sorry for delay. I can confirm issue is now resolved!

**Comment by alexandergoncharov** [comment]
@AMerkuri that's great! Thanks for keeping us in touch!

**Comment by miqmago** [comment]
It's happening the same with `1.12.0`

The strange thing is that it is not running on code-push update (it's running from binnary) and it's sending `Origin: null` anyway.

**Comment by miqmago** [comment]
Fixed, was using an old old old version of `cordova-plugin-ionic-webview` (1.3.1 or so).
Updated to 2.3.1 and started working again
